dataland-backend/src/test/kotlin/org/dataland/datalandbackend/ressources/StoredCompaniesRepositoriesTest.kt

```diff
@@ -0,0 +1,103 @@
+import org.dataland.datalandbackendutils.model.QaStatus
+import org.junit.jupiter.api.Test
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.jdbc.EmbeddedDatabaseConnection
+import org.springframework.boot.test.autoconfigure.jdbc.AutoConfigureTestDatabase
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest
+import org.springframework.boot.test.context.SpringBootTest
+
+@SpringBootTest(classes = [DatalandBackend::class], properties = ["spring.profiles.active=nodb"])
+
+@AutoConfigureTestDatabase(connection = EmbeddedDatabaseConnection.H2)
+class StoredCompaniesRepositoriesTest(
+    @Autowired val companyAlterationManager: CompanyAlterationManager,
+    @Autowired val dataMetaInformationManager: DataMetaInformationManager,
+@Autowired val companyQueryManager: CompanyQueryManager
+) {
+
+    @Test
+    fun `Check that only companies with accepted data sets are returned by the query`() {
+        val acceptedCompany = companyAlterationManager.addCompany(
+            CompanyInformation(
+                companyName = "TestName1",
+                headquarters = "TestHQ1",
+                identifiers = emptyMap(),
+                countryCode = "TestCountry",
+                isTeaserCompany = false,
+                companyAlternativeNames = listOf("alt1"),
+                sector = null,
+                companyLegalForm = null,
+                headquartersPostalCode = null,
+                website = null
+            )
+        )
+
+        dataMetaInformationManager.storeDataMetaInformation(DataMetaInformationEntity(
+            dataId = "1",
+            company = acceptedCompany,
+            dataType = "1",
+            uploaderUserId = "1",
+            uploadTime = 1,
+            reportingPeriod = "1",
+            currentlyActive = true,
+            qaStatus = QaStatus.Accepted
+        )
+        )
+
+        val pendingCompany = companyAlterationManager.addCompany(
+            CompanyInformation(
+                companyName = "TestName1",
+                headquarters = "TestHQ1",
+                identifiers = emptyMap(),
+                countryCode = "TestCountry",
+                isTeaserCompany = false,
+                companyAlternativeNames = listOf("alt1"),
+                sector = null,
+                companyLegalForm = null,
+                headquartersPostalCode = null,
+                website = null
+            )
+        )
+
+        dataMetaInformationManager.storeDataMetaInformation(DataMetaInformationEntity(
+            dataId = "2",
+            company = pendingCompany,
+            dataType = "2",
+            uploaderUserId = "2",
+            uploadTime = 2,
+            reportingPeriod = "2",
+            currentlyActive = null,
+            qaStatus = QaStatus.Pending
+        )
+        )
+
+
+
+        println(companyQueryManager.getCompanyById(acceptedCompany.companyId))
```

Comment:
Debugging statement?

Reply:
Yes. The file is gone now and the intent is implemented in an e2e test.

Reply:
ok